modules/olm-configuring-aws-sts.adoc

```diff
@@ -0,0 +1,138 @@
+
+* You have extracted and prepared the `ccoctl` binary during cluster installation. For more information, see "Configuring the Cloud Credential Operator utility".
+
+.Procedure
+
+. Determine the name of the `CredentialsRequest` custom resource (CR) in the `openshift-cloud-credential-operator` namespace for the Operator:
+//How do they do that?
+
+. Extract the CR to the `credrequests` directory by running the following command:
++
+[source,terminal]
+----
+$ oc get credentialsrequest \
+  -n openshift-cloud-credential-operator <custom_resource_name> \
+  -o yaml > <path_to_directory_with_list_of_credentials_requests>/credrequests/<custom_resource_name>.yaml
+----
+
+//Existing CSI Operator has an example CR, but the CR extraction should create a YAML file that is correct without the user needing to mess with it, right?
+////
+. Create and save an EFS `CredentialsRequest` YAML file, such as shown in the following example, and then place it in the `credrequests` directory:
++
+.Example
++
+[source, yaml]
+----
+apiVersion: cloudcredential.openshift.io/v1
+kind: CredentialsRequest
+metadata:
+  name: openshift-aws-efs-csi-driver
+  namespace: openshift-cloud-credential-operator
+spec:
+  providerSpec:
+    apiVersion: cloudcredential.openshift.io/v1
+    kind: AWSProviderSpec
+    statementEntries:
+    - action:
+      - elasticfilesystem:*
+      effect: Allow
+      resource: '*'
+  secretRef:
+    name: aws-efs-cloud-credentials
+    namespace: openshift-cluster-csi-drivers
+  serviceAccountNames:
+  - aws-efs-csi-driver-operator
+  - aws-efs-csi-driver-controller-sa
+----
+////
+
+. Use the `ccoctl` tool to process all `CredentialsRequest` objects in the `credrequests` directory by running the following command:
++
+[source, terminal]
+----
+$ ccoctl aws create-iam-roles \
+  --name=<name> \
+  --region=<aws_region> \
+  --credentials-requests-dir=<path_to_directory_with_list_of_credentials_requests>/credrequests \ --identity-provider-arn=arn:aws:iam::<aws_account_id>:oidc-provider/<name>-oidc.s3.<aws_region>.amazonaws.com
+----
++
+where:
++
+--
+** `<name>` is the name used to tag any cloud resources that are created for tracking.
+** `<aws-region>` is the AWS region in which cloud resources will be created.
+** `<path_to_ccoctl_output_dir>` is the path to the public key file that the `ccoctl aws create-key-pair` command generated.
+** `<aws_account_id>` is the AWS account ID.
+-- 
++
+The command creates an IAM role in AWS for the Operator and a corresponding `<operator_credentialsrequest_name>-credentials.yaml` file in the `<path_to_ccoctl_output_dir>/manifests/` directory of the local file system.
+
+//This step is not in the ALB proc or the existing STS content. Needs SME assessment.
+. Create the Operator credentials and secret by running the following command:
++
+[source, terminal]
+----
+$ oc create -f <path_to_ccoctl_output_dir>/manifests/<operator_credentialsrequest_name>-credentials.yaml
+----
```

Comment:
This step is not in the ALB proc or the existing STS content. Needs SME assessment.

Reply:
ACK, will leave the next one in since that's what we have elsewhere.